File: csharp/src/main/CsharpDotNet2/Org/OpenAPITools/Api/DefaultApi.cs
using System;
using System.Collections.Generic;
using RestSharp;
using Org.OpenAPITools.Client;
using com.spoonacular.client.model;

namespace com.spoonacular
{
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public interface IDefaultApi
    {
        /// <summary>
        /// Analyze Recipe This endpoint allows you to send raw recipe information, such as title, servings, and ingredients, to then see what we compute (badges, diets, nutrition, and more). This is useful if you have your own recipe data and want to enrich it with our semantic analysis.
        /// </summary>
        /// <param name="analyzeRecipeRequest">Example request body.</param>
        /// <param name="language">The input language, either \&quot;en\&quot; or \&quot;de\&quot;.</param>
        /// <param name="includeNutrition">Whether nutrition data should be added to correctly parsed ingredients.</param>
        /// <param name="includeTaste">Whether taste data should be added to correctly parsed ingredients.</param>
        /// <returns>Object</returns>
        Object AnalyzeRecipe (AnalyzeRecipeRequest analyzeRecipeRequest, string language, bool? includeNutrition, bool? includeTaste);
        /// <summary>
        /// Create Recipe Card Generate a recipe card for a recipe.
        /// </summary>
        /// <param name="id">The recipe id.</param>
        /// <param name="mask">The mask to put over the recipe image (\&quot;ellipseMask\&quot;, \&quot;diamondMask\&quot;, \&quot;starMask\&quot;, \&quot;heartMask\&quot;, \&quot;potMask\&quot;, \&quot;fishMask\&quot;).</param>
        /// <param name="backgroundImage">The background image (\&quot;none\&quot;,\&quot;background1\&quot;, or \&quot;background2\&quot;).</param>
        /// <param name="backgroundColor">The background color for the recipe card as a hex-string.</param>
        /// <param name="fontColor">The font color for the recipe card as a hex-string.</param>
        /// <returns>Object</returns>
        Object CreateRecipeCardGet (decimal? id, string mask, string backgroundImage, string backgroundColor, string fontColor);
        /// <summary>
        /// Search Restaurants Search through thousands of restaurants (in North America) by location, cuisine, budget, and more.
        /// </summary>
        /// <param name="query">The search query.</param>
        /// <param name="lat">The latitude of the user&#39;s location.</param>
        /// <param name="lng">The longitude of the user&#39;s location.\&quot;.</param>
        /// <param name="distance">The distance around the location in miles.</param>
        /// <param name="budget">The user&#39;s budget for a meal in USD.</param>
        /// <param name="cuisine">The cuisine of the restaurant.</param>
        /// <param name="minRating">The minimum rating of the restaurant between 0 and 5.</param>
        /// <param name="isOpen">Whether the restaurant must be open at the time of search.</param>
        /// <param name="sort">How to sort the results, one of the following &#39;cheapest&#39;, &#39;fastest&#39;, &#39;rating&#39;, &#39;distance&#39; or the default &#39;relevance&#39;.</param>
        /// <param name="page">The page number of results.</param>
        /// <returns>SearchRestaurants200Response</returns>
        SearchRestaurants200Response SearchRestaurants (string query, decimal? lat, decimal? lng, decimal? distance, decimal? budget, string cuisine, decimal? minRating, bool? isOpen, string sort, decimal? page);
    }

    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public class DefaultApi : IDefaultApi
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="DefaultApi"/> class.
        /// </summary>
        /// <param name="apiClient"> an instance of ApiClient (optional)</param>
        /// <returns></returns>
        public DefaultApi(ApiClient apiClient = null)
        {
            if (apiClient == null) // use the default one in Configuration
                this.ApiClient = Configuration.DefaultApiClient;
            else
                this.ApiClient = apiClient;
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="DefaultApi"/> class.
        /// </summary>
        /// <returns></returns>
        public DefaultApi(String basePath)
        {
            this.ApiClient = new ApiClient(basePath);
        }

        /// <summary>
        /// Sets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public void SetBasePath(String basePath)
        {
            this.ApiClient.BasePath = basePath;
        }

        /// <summary>
        /// Gets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public String GetBasePath(String basePath)
        {
            return this.ApiClient.BasePath;
        }

        /// <summary>
        /// Gets or sets the API client.
        /// </summary>
        /// <value>An instance of the ApiClient</value>
        public ApiClient ApiClient {get; set;}

        /// <summary>
        /// Analyze Recipe This endpoint allows you to send raw recipe information, such as title, servings, and ingredients, to then see what we compute (badges, diets, nutrition, and more). This is useful if you have your own recipe data and want to enrich it with our semantic analysis.
        /// </summary>
        /// <param name="analyzeRecipeRequest">Example request body.</param>
        /// <param name="language">The input language, either \&quot;en\&quot; or \&quot;de\&quot;.</param>
        /// <param name="includeNutrition">Whether nutrition data should be added to correctly parsed ingredients.</param>
        /// <param name="includeTaste">Whether taste data should be added to correctly parsed ingredients.</param>
        /// <returns>Object</returns>
        public Object AnalyzeRecipe (AnalyzeRecipeRequest analyzeRecipeRequest, string language, bool? includeNutrition, bool? includeTaste)
        {
            
            // verify the required parameter 'analyzeRecipeRequest' is set
            if (analyzeRecipeRequest == null) throw new ApiException(400, "Missing required parameter 'analyzeRecipeRequest' when calling AnalyzeRecipe");
            

            var path = "/recipes/analyze";
            path = path.Replace("{format}", "json");
            
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;

             if (language != null) queryParams.Add("language", ApiClient.ParameterToString(language)); // query parameter
 if (includeNutrition != null) queryParams.Add("includeNutrition", ApiClient.ParameterToString(includeNutrition)); // query parameter
 if (includeTaste != null) queryParams.Add("includeTaste", ApiClient.ParameterToString(includeTaste)); // query parameter
                                    postBody = ApiClient.Serialize(analyzeRecipeRequest); // http body (model) parameter

            // authentication setting, if any
            String[] authSettings = new String[] { "apiKeyScheme" };

            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.POST, queryParams, postBody, headerParams, formParams, fileParams, authSettings);

            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling AnalyzeRecipe: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling AnalyzeRecipe: " + response.ErrorMessage, response.ErrorMessage);

            return (Object) ApiClient.Deserialize(response.Content, typeof(Object), response.Headers);
        }

        /// <summary>
        /// Create Recipe Card Generate a recipe card for a recipe.
        /// </summary>
        /// <param name="id">The recipe id.</param>
        /// <param name="mask">The mask to put over the recipe image (\&quot;ellipseMask\&quot;, \&quot;diamondMask\&quot;, \&quot;starMask\&quot;, \&quot;heartMask\&quot;, \&quot;potMask\&quot;, \&quot;fishMask\&quot;).</param>
        /// <param name="backgroundImage">The background image (\&quot;none\&quot;,\&quot;background1\&quot;, or \&quot;background2\&quot;).</param>
        /// <param name="backgroundColor">The background color for the recipe card as a hex-string.</param>
        /// <param name="fontColor">The font color for the recipe card as a hex-string.</param>
        /// <returns>Object</returns>
        public Object CreateRecipeCardGet (decimal? id, string mask, string backgroundImage, string backgroundColor, string fontColor)
        {
            
            // verify the required parameter 'id' is set
            if (id == null) throw new ApiException(400, "Missing required parameter 'id' when calling CreateRecipeCardGet");
            

            var path = "/recipes/{id}/card";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "id" + "}", ApiClient.ParameterToString(id));

            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;

             if (mask != null) queryParams.Add("mask", ApiClient.ParameterToString(mask)); // query parameter
 if (backgroundImage != null) queryParams.Add("backgroundImage", ApiClient.ParameterToString(backgroundImage)); // query parameter
 if (backgroundColor != null) queryParams.Add("backgroundColor", ApiClient.ParameterToString(backgroundColor)); // query parameter
 if (fontColor != null) queryParams.Add("fontColor", ApiClient.ParameterToString(fontColor)); // query parameter
                                    
            // authentication setting, if any
            String[] authSettings = new String[] { "apiKeyScheme" };

            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);

            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling CreateRecipeCardGet: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling CreateRecipeCardGet: " + response.ErrorMessage, response.ErrorMessage);

            return (Object) ApiClient.Deserialize(response.Content, typeof(Object), response.Headers);
        }

        /// <summary>
        /// Search Restaurants Search through thousands of restaurants (in North America) by location, cuisine, budget, and more.
        /// </summary>
        /// <param name="query">The search query.</param>
        /// <param name="lat">The latitude of the user&#39;s location.</param>
        /// <param name="lng">The longitude of the user&#39;s location.\&quot;.</param>
        /// <param name="distance">The distance around the location in miles.</param>
        /// <param name="budget">The user&#39;s budget for a meal in USD.</param>
        /// <param name="cuisine">The cuisine of the restaurant.</param>
        /// <param name="minRating">The minimum rating of the restaurant between 0 and 5.</param>
        /// <param name="isOpen">Whether the restaurant must be open at the time of search.</param>
        /// <param name="sort">How to sort the results, one of the following &#39;cheapest&#39;, &#39;fastest&#39;, &#39;rating&#39;, &#39;distance&#39; or the default &#39;relevance&#39;.</param>
        /// <param name="page">The page number of results.</param>
        /// <returns>SearchRestaurants200Response</returns>
        public SearchRestaurants200Response SearchRestaurants (string query, decimal? lat, decimal? lng, decimal? distance, decimal? budget, string cuisine, decimal? minRating, bool? isOpen, string sort, decimal? page)
        {
            

            var path = "/food/restaurants/search";
            path = path.Replace("{format}", "json");
            
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;

             if (query != null) queryParams.Add("query", ApiClient.ParameterToString(query)); // query parameter
 if (lat != null) queryParams.Add("lat", ApiClient.ParameterToString(lat)); // query parameter
 if (lng != null) queryParams.Add("lng", ApiClient.ParameterToString(lng)); // query parameter
 if (distance != null) queryParams.Add("distance", ApiClient.ParameterToString(distance)); // query parameter
 if (budget != null) queryParams.Add("budget", ApiClient.ParameterToString(budget)); // query parameter
 if (cuisine != null) queryParams.Add("cuisine", ApiClient.ParameterToString(cuisine)); // query parameter
 if (minRating != null) queryParams.Add("min-rating", ApiClient.ParameterToString(minRating)); // query parameter
 if (isOpen != null) queryParams.Add("is-open", ApiClient.ParameterToString(isOpen)); // query parameter
 if (sort != null) queryParams.Add("sort", ApiClient.ParameterToString(sort)); // query parameter
 if (page != null) queryParams.Add("page", ApiClient.ParameterToString(page)); // query parameter
                                    
            // authentication setting, if any
            String[] authSettings = new String[] { "apiKeyScheme" };

            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);

            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling SearchRestaurants: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling SearchRestaurants: " + response.ErrorMessage, response.ErrorMessage);

            return (SearchRestaurants200Response) ApiClient.Deserialize(response.Content, typeof(SearchRestaurants200Response), response.Headers);
        }

    }
}

File: javascript/src/api/DefaultApi.js
/**
 * spoonacular API
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over thousands of recipes, thousands of ingredients, 800,000 food products, over 100,000 menu items, and restaurants. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 2.0.2
 * Contact: mail@spoonacular.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import AnalyzeRecipeRequest from '../model/AnalyzeRecipeRequest';
import SearchRestaurants200Response from '../model/SearchRestaurants200Response';

/**
* Default service.
* @module api/DefaultApi
* @version 2.0.2
*/
export default class DefaultApi {

    /**
    * Constructs a new DefaultApi. 
    * @alias module:api/DefaultApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the analyzeRecipe operation.
     * @callback module:api/DefaultApi~analyzeRecipeCallback
     * @param {String} error Error message, if any.
     * @param {Object} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Analyze Recipe
     * This endpoint allows you to send raw recipe information, such as title, servings, and ingredients, to then see what we compute (badges, diets, nutrition, and more). This is useful if you have your own recipe data and want to enrich it with our semantic analysis.
     * @param {module:model/AnalyzeRecipeRequest} analyzeRecipeRequest Example request body.
     * @param {Object} opts Optional parameters
     * @param {String} [language] The input language, either \"en\" or \"de\".
     * @param {Boolean} [includeNutrition] Whether nutrition data should be added to correctly parsed ingredients.
     * @param {Boolean} [includeTaste] Whether taste data should be added to correctly parsed ingredients.
     * @param {module:api/DefaultApi~analyzeRecipeCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link Object}
     */
    analyzeRecipe(analyzeRecipeRequest, opts, callback) {
      opts = opts || {};
      let postBody = analyzeRecipeRequest;
      // verify the required parameter 'analyzeRecipeRequest' is set
      if (analyzeRecipeRequest === undefined || analyzeRecipeRequest === null) {
        throw new Error("Missing the required parameter 'analyzeRecipeRequest' when calling analyzeRecipe");
      }

      let pathParams = {
      };
      let queryParams = {
        'language': opts['language'],
        'includeNutrition': opts['includeNutrition'],
        'includeTaste': opts['includeTaste']
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['apiKeyScheme'];
      let contentTypes = ['application/json'];
      let accepts = ['application/json'];
      let returnType = Object;
      return this.apiClient.callApi(
        '/recipes/analyze', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the createRecipeCardGet operation.
     * @callback module:api/DefaultApi~createRecipeCardGetCallback
     * @param {String} error Error message, if any.
     * @param {Object} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Create Recipe Card
     * Generate a recipe card for a recipe.
     * @param {Number} id The recipe id.
     * @param {Object} opts Optional parameters
     * @param {String} [mask] The mask to put over the recipe image (\"ellipseMask\", \"diamondMask\", \"starMask\", \"heartMask\", \"potMask\", \"fishMask\").
     * @param {String} [backgroundImage] The background image (\"none\",\"background1\", or \"background2\").
     * @param {String} [backgroundColor] The background color for the recipe card as a hex-string.
     * @param {String} [fontColor] The font color for the recipe card as a hex-string.
     * @param {module:api/DefaultApi~createRecipeCardGetCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link Object}
     */
    createRecipeCardGet(id, opts, callback) {
      opts = opts || {};
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling createRecipeCardGet");
      }

      let pathParams = {
        'id': id
      };
      let queryParams = {
        'mask': opts['mask'],
        'backgroundImage': opts['backgroundImage'],
        'backgroundColor': opts['backgroundColor'],
        'fontColor': opts['fontColor']
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['apiKeyScheme'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = Object;
      return this.apiClient.callApi(
        '/recipes/{id}/card', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the searchRestaurants operation.
     * @callback module:api/DefaultApi~searchRestaurantsCallback
     * @param {String} error Error message, if any.
     * @param {module:model/SearchRestaurants200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Search Restaurants
     * Search through thousands of restaurants (in North America) by location, cuisine, budget, and more.
     * @param {Object} opts Optional parameters
     * @param {String} [query] The search query.
     * @param {Number} [lat] The latitude of the user's location.
     * @param {Number} [lng] The longitude of the user's location.\".
     * @param {Number} [distance] The distance around the location in miles.
     * @param {Number} [budget] The user's budget for a meal in USD.
     * @param {String} [cuisine] The cuisine of the restaurant.
     * @param {Number} [minRating] The minimum rating of the restaurant between 0 and 5.
     * @param {Boolean} [isOpen] Whether the restaurant must be open at the time of search.
     * @param {String} [sort] How to sort the results, one of the following 'cheapest', 'fastest', 'rating', 'distance' or the default 'relevance'.
     * @param {Number} [page] The page number of results.
     * @param {module:api/DefaultApi~searchRestaurantsCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/SearchRestaurants200Response}
     */
    searchRestaurants(opts, callback) {
      opts = opts || {};
      let postBody = null;

      let pathParams = {
      };
      let queryParams = {
        'query': opts['query'],
        'lat': opts['lat'],
        'lng': opts['lng'],
        'distance': opts['distance'],
        'budget': opts['budget'],
        'cuisine': opts['cuisine'],
        'min-rating': opts['minRating'],
        'is-open': opts['isOpen'],
        'sort': opts['sort'],
        'page': opts['page']
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['apiKeyScheme'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = SearchRestaurants200Response;
      return this.apiClient.callApi(
        '/food/restaurants/search', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
